src/gui/transferlistwidget.cpp

```diff
@@ -490,58 +490,43 @@ void TransferListWidget::previewSelectedTorrents()
     }
 }
 
-void TransferListWidget::setDlLimitSelectedTorrents()
+void TransferListWidget::setSpeedLimitsSelectedTorrents()
 {
-    QVector<BitTorrent::TorrentHandle *> torrentsList;
-    for (BitTorrent::TorrentHandle *const torrent : asConst(getSelectedTorrents())) {
-        if (torrent->isSeed())
-            continue;
-        torrentsList += torrent;
-    }
+    const QVector<BitTorrent::TorrentHandle *> torrentsList = getSelectedTorrents();
     if (torrentsList.empty()) return;
 
-    int oldLimit = torrentsList.first()->downloadLimit();
-    for (BitTorrent::TorrentHandle *const torrent : asConst(torrentsList)) {
-        if (torrent->downloadLimit() != oldLimit) {
-            oldLimit = -1;
+    int oldUploadLimit = torrentsList.first()->uploadLimit();
+    int oldDownloadLimit = torrentsList.first()->downloadLimit();
+
+    for (const BitTorrent::TorrentHandle *torrent : torrentsList) {
+        if (torrent->uploadLimit() != oldUploadLimit) {
+            oldUploadLimit = -1;
             break;
         }
     }
 
-    bool ok = false;
-    const long newLimit = SpeedLimitDialog::askSpeedLimit(
-                this, &ok, tr("Torrent Download Speed Limiting"), oldLimit
-                , BitTorrent::Session::instance()->globalDownloadSpeedLimit());
-    if (!ok) return;
-
-    for (BitTorrent::TorrentHandle *const torrent : asConst(torrentsList)) {
-        qDebug("Applying download speed limit of %ld Kb/s to torrent %s", (newLimit / 1024l), qUtf8Printable(torrent->hash()));
-        torrent->setDownloadLimit(newLimit);
-    }
-}
-
-void TransferListWidget::setUpLimitSelectedTorrents()
-{
-    QVector<BitTorrent::TorrentHandle *> torrentsList = getSelectedTorrents();
-    if (torrentsList.empty()) return;
-
-    int oldLimit = torrentsList.first()->uploadLimit();
-    for (BitTorrent::TorrentHandle *const torrent : asConst(torrentsList)) {
-        if (torrent->uploadLimit() != oldLimit) {
-            oldLimit = -1;
+    for (const BitTorrent::TorrentHandle *torrent : torrentsList) {
+        if (torrent->downloadLimit() != oldDownloadLimit) {
+            oldDownloadLimit = -1;
             break;
         }
     }
 
-    bool ok = false;
-    const long newLimit = SpeedLimitDialog::askSpeedLimit(
-                this, &ok, tr("Torrent Upload Speed Limiting"), oldLimit
-                , BitTorrent::Session::instance()->globalUploadSpeedLimit());
+    const BitTorrent::Session *session = BitTorrent::Session::instance();
+
+    SpeedLimits speedLimits = {
+        oldUploadLimit,
+        oldDownloadLimit,
+        session->isAltGlobalSpeedLimitEnabled() ? session->altGlobalUploadSpeedLimit() : session->globalUploadSpeedLimit(),
```

Comment:
add a `const bool isAltLimitEnabled = session->isAltGlobalSpeedLimitEnabled();` under `session` variable and use it here.

Reply:
done.